// forta-agent/dist/sdk/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.verifyJwt = exports.decodeJwt = exports.fetchJwt = exports.getAlerts = exports.getTransactionReceipt = exports.configureContainer = exports.isPrivateFindings = exports.setPrivateFindings = exports.keccak256 = exports.ethers = exports.getEthersBatchProvider = exports.getEthersProvider = exports.createAlertEvent = exports.createBlockEvent = exports.createTransactionEvent = exports.getJsonRpcUrl = exports.Network = exports.EventType = exports.AlertEvent = exports.TransactionEvent = exports.BlockEvent = exports.EntityType = exports.Label = exports.FindingType = exports.FindingSeverity = exports.Finding = void 0;
var ethers_1 = require("ethers");
Object.defineProperty(exports, "ethers", { enumerable: true, get: function () { return ethers_1.ethers; } });
var finding_1 = require("./finding");
Object.defineProperty(exports, "Finding", { enumerable: true, get: function () { return finding_1.Finding; } });
Object.defineProperty(exports, "FindingSeverity", { enumerable: true, get: function () { return finding_1.FindingSeverity; } });
Object.defineProperty(exports, "FindingType", { enumerable: true, get: function () { return finding_1.FindingType; } });
var label_1 = require("./label");
Object.defineProperty(exports, "Label", { enumerable: true, get: function () { return label_1.Label; } });
Object.defineProperty(exports, "EntityType", { enumerable: true, get: function () { return label_1.EntityType; } });
var block_event_1 = require("./block.event");
Object.defineProperty(exports, "BlockEvent", { enumerable: true, get: function () { return block_event_1.BlockEvent; } });
var alert_event_1 = require("./alert.event");
Object.defineProperty(exports, "AlertEvent", { enumerable: true, get: function () { return alert_event_1.AlertEvent; } });
var transaction_event_1 = require("./transaction.event");
Object.defineProperty(exports, "TransactionEvent", { enumerable: true, get: function () { return transaction_event_1.TransactionEvent; } });
var utils_1 = require("./utils");
Object.defineProperty(exports, "createBlockEvent", { enumerable: true, get: function () { return utils_1.createBlockEvent; } });
Object.defineProperty(exports, "createTransactionEvent", { enumerable: true, get: function () { return utils_1.createTransactionEvent; } });
Object.defineProperty(exports, "createAlertEvent", { enumerable: true, get: function () { return utils_1.createAlertEvent; } });
Object.defineProperty(exports, "getJsonRpcUrl", { enumerable: true, get: function () { return utils_1.getJsonRpcUrl; } });
Object.defineProperty(exports, "getEthersProvider", { enumerable: true, get: function () { return utils_1.getEthersProvider; } });
Object.defineProperty(exports, "getEthersBatchProvider", { enumerable: true, get: function () { return utils_1.getEthersBatchProvider; } });
Object.defineProperty(exports, "keccak256", { enumerable: true, get: function () { return utils_1.keccak256; } });
Object.defineProperty(exports, "setPrivateFindings", { enumerable: true, get: function () { return utils_1.setPrivateFindings; } });
Object.defineProperty(exports, "isPrivateFindings", { enumerable: true, get: function () { return utils_1.isPrivateFindings; } });
Object.defineProperty(exports, "getTransactionReceipt", { enumerable: true, get: function () { return utils_1.getTransactionReceipt; } });
Object.defineProperty(exports, "getAlerts", { enumerable: true, get: function () { return utils_1.getAlerts; } });
Object.defineProperty(exports, "fetchJwt", { enumerable: true, get: function () { return utils_1.fetchJwt; } });
Object.defineProperty(exports, "decodeJwt", { enumerable: true, get: function () { return utils_1.decodeJwt; } });
Object.defineProperty(exports, "verifyJwt", { enumerable: true, get: function () { return utils_1.verifyJwt; } });
var di_container_1 = __importDefault(require("../cli/di.container"));
var configureContainer = function (args) {
    if (args === void 0) { args = {}; }
    return (0, di_container_1.default)(args);
};
exports.configureContainer = configureContainer;
var EventType;
(function (EventType) {
    EventType[EventType["BLOCK"] = 0] = "BLOCK";
    EventType[EventType["REORG"] = 1] = "REORG";
})(EventType || (EventType = {}));
exports.EventType = EventType;
var Network;
(function (Network) {
    Network[Network["MAINNET"] = 1] = "MAINNET";
    Network[Network["ROPSTEN"] = 3] = "ROPSTEN";
    Network[Network["RINKEBY"] = 4] = "RINKEBY";
    Network[Network["GOERLI"] = 5] = "GOERLI";
    Network[Network["POLYGON"] = 137] = "POLYGON";
    Network[Network["BSC"] = 56] = "BSC";
    Network[Network["AVALANCHE"] = 43114] = "AVALANCHE";
    Network[Network["ARBITRUM"] = 42161] = "ARBITRUM";
    Network[Network["OPTIMISM"] = 10] = "OPTIMISM";
    Network[Network["FANTOM"] = 250] = "FANTOM";
})(Network || (Network = {}));
exports.Network = Network;
